src/main/java/com/palantir/gradle/versions/FixLegacyJavaConfigurationsPlugin.java

```diff
@@ -44,41 +44,30 @@ public final void apply(Project project) {
             return;
         }
 
-        Configuration unifiedClasspath = project
-                .getRootProject()
+        Configuration unifiedClasspath = project.getRootProject()
                 .getConfigurations()
                 .findByName(VersionsLockPlugin.UNIFIED_CLASSPATH_CONFIGURATION_NAME);
         Preconditions.checkNotNull(
-                unifiedClasspath,
-                "FixLegacyJavaConfigurationsPlugin must be applied after VersionsLockPlugin");
+                unifiedClasspath, "FixLegacyJavaConfigurationsPlugin must be applied after VersionsLockPlugin");
 
         fixLegacyResolvableJavaConfigurations(project, unifiedClasspath);
     }
 
     private void fixLegacyResolvableJavaConfigurations(Project project, Configuration unifiedClasspath) {
-        Stream.of(
-                JavaPlugin.COMPILE_CONFIGURATION_NAME,
-                JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME,
-                JavaPlugin.RUNTIME_CONFIGURATION_NAME)
+        Stream.of(JavaPlugin.COMPILE_CONFIGURATION_NAME, JavaPlugin.COMPILE_ONLY_CONFIGURATION_NAME, JavaPlugin
+                .RUNTIME_CONFIGURATION_NAME)
```

Comment:
Currently it will try to lay out all arguments on the 2nd line if possible, but if not, it will prefer breaking the last argument in certain cases (if it's a lambda, method args or generally anything that includes a dot).
This can probably be made stricter so that field accesses like this one aren't allowed to be broken this way.

Reply:
why wouldn't this render as `\nJavaPlugin.RUNTIME_CONFIGURATION_NAME`?

Reply:
Sorry I missed this. It won't ever do that, as the three parameters are all in a single "level", separated by breaks that are either taken all as a group or none at all.

As for why it accepts splitting there, that's due to an overly lenient last-argument-splitting heuristic that's now been fixed.
See https://github.com/palantir/gradle-consistent-versions/pull/305/files#diff-56e0cf074cb9bc301934d0fd45ed3508R56-R59